### PR TITLE
Create expo-app-auth deprecated docs

### PR DESCRIPTION
**Pull request by EvanBacon**
Add a guide for migrating off of expo-app-auth and onto expo-auth-session.

### CODE REVIEW

File: expo-app-auth-deprecated.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Deprecating `expo-app-auth`
+
+In May 2020 we rewrote [`expo-auth-session`](https://docs.expo.dev/versions/latest/sdk/auth-session/) as a pure JS replacement for most of the authentication systems in Expo.
+
+**AuthSession** solves the following issues with `expo-app-auth`:
+
+- Many authentication providers require nonstandard customizations which weren't possible with the existing native API.
+- `expo-app-auth` includes heavy native packages which weren't fully exposed, meaning the builds were needlessly larger.
```
Comment (brentvatne) [inline expo-app-auth-deprecated.md:8]:
```suggestion
- `expo-app-auth` includes heavy native packages which weren't fully exposed, resulting in larger than necessary build sizes.
```

File: expo-app-auth-deprecated.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+- Debugging `expo-app-auth` proved to be very difficult.
+- `expo-app-auth` didn't include web support and the native interface wasn't very cross platform.
```
Comment (brentvatne) [inline expo-app-auth-deprecated.md:10]:
```suggestion
- `expo-app-auth` didn't include web support and the native interface wasn't cross platform.
```

File: expo-app-auth-deprecated.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+- `expo-app-auth` extends a native library from Google which mostly just worked with the Google authentication.
+
+## Migration
+
+Many auth providers have robust documentation for integrating with `expo-auth-session` making it easier to move over: [Learn more](https://docs.expo.dev/guides/authentication/).
+
+You will probably want to migrate to the Hooks API provided by `expo-auth-session` which has no comparable API in `expo-app-auth`, but if you'd like to migrate one to one, here is the closest estimation:
```
Comment (brentvatne) [inline expo-app-auth-deprecated.md:17]: "here is the closest estimation:" - missing something here?

File: expo-app-auth-deprecated.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+### `issuer`
+
+An "issuer" is a URL to a server that returns an object of URLs that are used for interacting with the authentication service. For example, Google uses `https://accounts.google.com/.well-known/openid-configuration`.
```
Comment (brentvatne) [inline expo-app-auth-deprecated.md:21]:
```suggestion
An "issuer" is a URL for a server that returns an object of URLs that are used for interacting with the authentication service. For example, Google uses `https://accounts.google.com/.well-known/openid-configuration`.
```

File: expo-app-auth-deprecated.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+  // The discovery goes from null to an object when the request finishes.
+
+  return null;
+}
+```
+
+### `AppAuth.authAsync`
+
+The `authAsync` method contains a lot of complexity for things like auto exchanging client side secrets (never store your secrets in the client!). This functionality has been divided across a couple of different methods and classes in `expo-auth-session`. It is highly recommended that you use the React Hooks API for performing authentication in `expo-auth-session` via `AuthSession.useAuthRequest()`.
```
Comment (brentvatne) [inline expo-app-auth-deprecated.md:63]:
```suggestion
The `authAsync` method encapsulates a lot of functionality, such as auto exchanging client side secrets (never store your secrets in the client!). This functionality has been divided across a couple of different methods and classes in `expo-auth-session`. It is highly recommended that you use the React Hooks API for performing authentication in `expo-auth-session` via `AuthSession.useAuthRequest()`.
```

File: expo-app-auth-deprecated.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+There is currently no replacement for this iOS-only constant which returns the URI schemes from the app's native `Info.plist`. This may be added to `expo-linking` or `expo-application` in the future.
+
+Check out the [redirect URI patterns](https://docs.expo.dev/guides/authentication/#redirect-uri-patterns) guide for more info.
+
+### `AppAuth.OAuthRedirect`
+
+There is currently no replacement for this constant which simply indicates a native URI scheme that maps to the bundle identifier on iOS and the package name on Android.
+
+This can be generally retrieved by using `expo-application`:
```
Comment (brentvatne) [inline expo-app-auth-deprecated.md:149]: why "generally"? what exceptions are there?

**Review by brentvatne — APPROVED** [review]

File: expo-app-auth-deprecated.md (hunk 6 above)
Comment (EvanBacon) [reply expo-app-auth-deprecated.md:149]: Android package name ≠ applicationId but we use the package name for the uri scheme.

**Pull request merged**